### PR TITLE
[c4core] Update submodules version 

### PR DESCRIPTION
**Pull request by MichaelGoulding**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [ ] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [ ] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by FrankXie05** [comment]
`CI` error form `config-x86-windows-out.log`
```
CMake Error at cmake/c4Project.cmake:1042 (find_program):
  Could not find GIT using the following names: git
Call Stack (most recent call first):
  cmake/c4Project.cmake:884 (c4_proj_get_version)
  cmake/c4Project.cmake:825 (c4_set_default_pack_properties)
  CMakeLists.txt:110 (c4_pack_project)
```
[config-x86-windows-out.log](https://github.com/microsoft/vcpkg/files/13596321/config-x86-windows-out.log)

**Comment by MichaelGoulding** [comment]
Thank you for the help!

Thanks,
    Michael
________________________________

**Comment by dg0yt** [comment]
> It looks like there is no problem with the cmake command.: [`find_program(GIT git REQUIRED)`](https://github.com/biojppm/cmake/blob/9059c98181f944aec83b90a677c96f3b43bd1047/c4Project.cmake#L1181C30-L1181C35)

If a build needs git, it has to be provided via `vcpkg_find_acquire_program`, and forward to the build system via options or PATH.

But don't do this blindly. git shouldn't be used directly to clone extra sources. This breaks asset caching. Check what the build tries to do here. Maybe additional sources must be fetched by the portfile.

File: ports/c4core/portfile.cmake [hunk 1]
```diff
@@ -9,14 +9,14 @@ vcpkg_from_github(
     HEAD_REF master
 )
 
-set(CM_COMMIT_HASH fe41e86552046c3df9ba73a40bf3d755df028c1e)
+set(CM_COMMIT_HASH 95b2410e31ebf28b56a4fffffef52c7d13d657ad)
```
Comment (MichaelGoulding) [inline ports/c4core/portfile.cmake:12]:
```suggestion
set(CM_COMMIT_HASH 95b2410e31ebf28b56a4fffffef52c7d13d657ad)
set(CM_COMMIT_HASH fe41e86552046c3df9ba73a40bf3d755df028c1e)
```

**Review by BillyONeal — APPROVED** [review]

**Pull request merged**